### PR TITLE
update input credential process with prompt

### PR DESCRIPTION
**Pull request by zhangzcjames**

### CODE REVIEW

File: R/get_connection.R [hunk 1]
```diff
@@ -88,7 +88,7 @@ get_connection <- function(...) {
         username = cred[1]
         password = cred[2]
       }, error = function(e) {
-        stop("You are not using a Nuvolos application. Please provide both username and password arguments to establish a connection. In your R's console, please add your credentials by this command: input_nuvolos_credential('your_user_name', 'your_password')")
+        stop("You are not using a Nuvolos application. Please provide both username and password arguments to establish a connection. In your R's console, please add your credentials by this command: input_nuvolos_credential()")
```
Comment (andrewsali) [inline R/get_connection.R]: Why not directly call input_nuvolos_credential for the user? Instead of throwing an error?

File: R/get_connection.R [hunk 2]
```diff
@@ -172,15 +172,29 @@ get_nuvolos_db_path <- function() {
 #' 
 #' Using outside of Nuvolos only. This helps the user to store credentials safely at local device.
 #' @export
-input_nuvolos_credential <- function(username, password){
+input_nuvolos_credential <- function(){
```
Comment (andrewsali) [inline R/get_connection.R:176]: I think we should remove this whole part and use rstudioapi::askForSecret("Please input your Nuvolos username:") directly in the connection setup. So no need for a separate 'config setup', instead the user can simply click "save in keyring" for themselves.

File: DESCRIPTION [hunk 3]
```diff
@@ -9,4 +9,4 @@ License: MIT
 Encoding: UTF-8
 LazyData: true
 RoxygenNote: 7.0.2
-Imports: DBI, ini, odbc, rstudioapi
+Imports: DBI, ini, odbc, rstudioapi, keyring
```
Comment (andrewsali) [inline DESCRIPTION]: This violates lexicographic ordering

**Review by andrewsali — CHANGES_REQUESTED** [review]
We should remove the initial credential setup part as it's unnecessary.

File: R/get_connection.R (hunk 2 above)
Comment (zhangzcjames) [reply R/get_connection.R:176]: but what if they input a wrong one, and need to change it...

Comment (andrewsali) [reply R/get_connection.R:176]: I would assume that is handled by Rstudio :)

Comment (zhangzcjames) [reply R/get_connection.R:176]: No, I mean if the user inputs the credential incorrectly, the keyring will save it anyway.  Then the connection will always fail without an approach to reinput... Rstudio couldn't handle that either...

File: R/get_connection.R [hunk 4]
```diff
@@ -172,15 +173,29 @@ get_nuvolos_db_path <- function() {
...
   # store username & password
+  username <- rstudioapi::askForSecret("Please input your Nuvolos username:")
   keyring::key_set_with_value("nuvolos", "username", username)
+  
+  password <- rstudioapi::askForSecret("Please input your Nuvolos password:")
```
Comment (andrewsali) [inline R/get_connection.R]: don't you need to specify that this is the message /  title?

https://www.rdocumentation.org/packages/rstudioapi/versions/0.11/topics/askForSecret

Comment (zhangzcjames) [reply R/get_connection.R]: change it now. thx for catching this.

**Comment by zhangzcjames** [comment]
> shouldn't name be 'username'?

I change it now, although it doesn't matter... the keyring uses a separate "name" to record the username and password

File: R/get_connection.R [hunk 5]
```diff
@@ -172,15 +173,29 @@ get_nuvolos_db_path <- function() {
 #' 
 #' Using outside of Nuvolos only. This helps the user to store credentials safely at local device.
 #' @export
-input_nuvolos_credential <- function(username, password){
+input_nuvolos_credential <- function(){
   # store username & password
+  username <- rstudioapi::askForSecret("Please input your Nuvolos username:")
```
Comment (andrewsali) [inline R/get_connection.R]: Don't these also need updating @IronManJames ?

Comment (zhangzcjames) [reply R/get_connection.R]: oh, my bad, updated it....

**Pull request merged**